### DISCUSSION TITLE
Using HasFlag in an Entity Framework query

### DISCUSSION THREAD

**Discussion by YousefAK009** [alirezanet/InfiniteEnumFlags#14] [Q&A] [answered]
Can use HasFlag in EF query?

```c#

query.Where(x => x.Permission.HasFlag(Permissions.AllowEdit)).ToList();

```

**Comment by alirezanet** [comment]


**Reply by YousefAK009** [reply]
Hi @alirezanet,

Thanks for your response.

how will this work if the permission field may be encoded with multiple permissions?

For example, if I have the following permission field:

```c#

public class Permissions : InfiniteEnum<Permissions>
{
    public static readonly Flag<Permissions> None = new(-1); 
    public static readonly Flag<Permissions> AllowEdit = new(0);
    public static readonly Flag<Permissions> AllowDelete = new(1);
    public static readonly Flag<Permissions> AllowView = new(2);
}

public class User
{
    public int Id {get; set}
    public Permissions Permissions {get; set; }
}


user.Permissions = Permissions.AllowEdit | Permissions.AllowDelete | Permissions.AllowView;
```

How can I use a custom operator to filter the results to records with the permission "AllowEdit" or "AllowView"?

**Reply by alirezanet** [reply, marked as answer]
Hello @yousefak007,

Apologies for the confusion earlier. I initially misunderstood your question, thinking it was related to my other open-source project, Gridify 🫠. Please disregard my previous response. (I deleted it)

Regarding the database side, using HasFlag directly is not feasible because your database doesn't inherently support custom enum functionality. You can, however, retrieve the relevant data into memory and then check permissions there.

If you're working with a SQL Server database, you might consider creating a custom function in C# and incorporating it into your database. Keep in mind that this approach could potentially impact query performance.

A bit of context: InfiniteEnumFlags generates a unique hash-like string, making it non-queryable in the database. But with the right design, you may find that you don't actually need it. If you can share why you require this feature, I may be able to suggest a more suitable solution.

Ultimately, if you need to query permissions in the database, I recommend creating a separate table to store the permissions.

**Reply by alirezanet** [reply]
Looking at your example, I think you want to check if the user has required permissions or not, 
so you can fetch the user data from the database (which usually contains their permission uniqueId in a simple string field) and this way you can check if the user has access to a specific feature or not.

in the example I provided [here](https://github.com/alirezanet/InfiniteEnumFlags?tab=readme-ov-file#example), we're using the same flow.

**Reply by YousefAK009** [reply]
thanks a lot
